ice40/devices/top-routing-virt/CMakeLists.txt

```diff
@@ -1,4 +1,45 @@
 add_subdirectory(tiles)
 
 add_file_target(FILE arch.xml SCANNER_TYPE xml)
-define_device_type(DEVICE_TYPE top-routing-virt ARCH ice40 ARCH_XML arch.xml)
+
+get_target_property_required(CONDA_DIR env CONDA_DIR)
+get_target_property_required(PYTHON3 env PYTHON3)
+get_target_property_required(PYTHON3_TARGET env PYTHON3_TARGET)
+get_target_property_required(ICESTORM_TARGET env ICESTORM_TARGET)
+
+set(ICEBOX_TIMING ${symbiflow-arch-defs_SOURCE_DIR}/third_party/icestorm/icefuzz/timings.py)
+set(CLEAN_ICEBOX_TIMING cleaned_timing.txt)
+set(TIMING_TXT_FILE ${ICEBOX_SHARE}/timings_hx1k.txt)
+set(SDF_FILE timings_hx1k.sdf)
+
+add_custom_command(
+  OUTPUT ${CLEAN_ICEBOX_TIMING}
```

Comment:
Rather than two add_custom_command, might be better to just have one?

Reply:
You mean combine these? I think that just makes it more opaque.

Reply:
If you don't combine these, then you have to an introduce a target between these two add_custom_command.  Parallel CMake will attempt to do the grep step more than once if multiple targets need the sdf_target.  If you want to keep them as seperate add_custom_command, there needs to be an intermediate target between the two.  See https://github.com/SymbiFlow/symbiflow-arch-defs/pull/773/ and https://github.com/SymbiFlow/symbiflow-arch-defs/pull/825

CMake is pretty dumb in this regard.

Reply:
Done